### PR TITLE
Notifications > Comment Details: show new details view for cached comment

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #17790 

This is the first step in showing the [new](https://github.com/wordpress-mobile/WordPress-iOS/issues/17087) Comment Details view for a Comment Notification. Specifically, when a comment notification is selected, if the comment is already cached, it will be displayed with the new view. 

Technical details: because `CommentDetailViewController` requires a `Comment` object, the `Comment` must first be loaded with information from the `Notification`. `NotificationCommentDetailCoordinator` has been added to facilitate comment loading, as well as other Notification specific functionality that will be added later.

Of course since this is the first step in showing this view, there are a number of outstanding tasks, which are listed in the referenced issue. 

To test:
- Enable the `notificationCommentDetails` feature.
- To cache comments, go to My Site > Comments. Pick a site that corresponds to the Notifications you'll be testing.
- Go to Notifications > Comments.
- Select a notification.
- If the comment is cached, the new view will be displayed.
- If the comment is not cached, the old view will be displayed. This is temporary. Fetching non-cached comments will be added later. 
- All the functionality in this view should work as expected (Moderation, Edit, Reply, Like, etc.), just like from My Site > Comments > comment details. 

| Cached / New | Not cached / Old |
|--------|-------|
| new | old |

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete and disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is incomplete and disabled.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is incomplete and disabled.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift [hunk 1]
```diff
@@ -723,7 +724,24 @@ extension NotificationsViewController {
...
+                // If the comment is not cached, fall back to showing the old comment view.
+                // This is temporary until NotificationCommentDetailCoordinator can fetch the comment from the endpoint.
+
+                if let commentDetailViewController = notificationCommentDetailCoordinator.viewController {
+                    commentDetailViewController.navigationItem.largeTitleDisplayMode = .never
+                    self.showDetailViewController(commentDetailViewController, sender: nil)
+                } else {
+                    // TODO: remove when NotificationCommentDetailCoordinator updated to fetch comment.
+                    self.performSegue(withIdentifier: NotificationDetailsViewController.classNameWithoutNamespaces(), sender: note)
+                }
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift:741]: Nitpick, but as this method is already pretty long, could we extract these into a `presentCommentDetail(for: note)` method or something?

**Review by frosty — APPROVED** [review]
Looks good! Just one very minor comment, otherwise :shipit:

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift:741]: Sure. I'll do it in the next PR.

**Comment by ScoutHarris** [comment]
Thanks @frosty !

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17850-47f97e7</code>. IPA is available <a href='https://appcenter-filemanagement-distrib1ede6f06e.azureedge.net/47c84c26-d3a2-41d1-9a48-5fddb94c8d90/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=nZGo64e3MeGCrbFeAPwpmmVI1%2Bvn8IF7RNz9vjqP8Qg%3D&se=2022-02-02T18%3A07%3A36Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Pull request merged**